### PR TITLE
case statements

### PR DESCRIPTION
**Pull request by ntwb**
Some stats:

`case string :`- (space before colon)
```shell
ag "case\s[\'\w\']+\s\:" --stats-only
299 matches
47 files contained matches
1767 files searched
```

`case string:`- (no space before colon)
```shell
ag "case\s[\'\w\']+\:" --stats-only
2637 matches
148 files contained matches
1767 files searched
```

`case string ;`-  (space before semicolon)
```shell
ag "case\s[\'\w\']+\s\;" --stats-only
0 matches
0 files contained matches
1767 files searched
```

`case string ;`- (no space before semicolon)
```shell
ag "case\s[\'\w\']+\;" --stats-only
2 matches
2 files contained matches
1767 files searched
```

### CODE REVIEW

File: php-coding-standards.md [hunk 1]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
 foreach ( $foo as $bar ) { ...
 ```
 
+When using case statements, do it like so:
+
+```php
+
+switch ($i) {
+    case 1:
+        echo "i equals 1";
```
Comment (GaryJones) [inline php-coding-standards.md]: Single quotes please.

File: php-coding-standards.md [hunk 2]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
 foreach ( $foo as $bar ) { ...
 ```
 
+When using case statements, do it like so:
+
+```php
+
+switch ($i) {
```
Comment (GaryJones) [inline php-coding-standards.md]: Space around variable.

File: php-coding-standards.md [hunk 3]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
...
+    case 1:
+        echo "i equals 1";
+        break;
+    case 2:
+        echo "i equals 2";
```
Comment (GaryJones) [inline php-coding-standards.md]: Single quotes.

File: php-coding-standards.md [hunk 4]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
...
+        break;
+    case 3 : // incorrect, do not include a space before the colon
```
Comment (GaryJones) [inline php-coding-standards.md]: Capital `I`, and finish with full stop.

File: php-coding-standards.md [hunk 5]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
...
+        echo "i equals 2";
+        break;
+    case 4; // incorrect, use colons not semicolons
```
Comment (GaryJones) [inline php-coding-standards.md]: Capital `I` and finish with full stop.

File: php-coding-standards.md [hunk 6]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
...
+
+switch ($i) {
+    case 1:
+        echo "i equals 1";
+        break;
+    case 2:
+        echo "i equals 2";
+        break;
+    case 3 : // incorrect, do not include a space before the colon
+        echo "i equals 2";
```
Comment (GaryJones) [inline php-coding-standards.md]: Single quotes.

File: php-coding-standards.md [hunk 7]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
...
+        break;
+    case 4; // incorrect, use colons not semicolons
+        echo "i equals 2";
```
Comment (GaryJones) [inline php-coding-standards.md]: Single quotes.

File: php-coding-standards.md [hunk 8]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
...
+        break;
+    default:
+       echo "i is not equal to 0, 1 or 2";
```
Comment (GaryJones) [inline php-coding-standards.md]: Single quotes.

File: php-coding-standards.md [hunk 9]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
 foreach ( $foo as $bar ) { ...
 ```
 
+When using case statements, do it like so:
```
Comment (GaryJones) [inline php-coding-standards.md]: case => `case`.

**Comment by ntwb** [comment]
Ha, the irony is strong in this original PR 😝 

Updated and fixed 😏

File: php-coding-standards.md [hunk 10]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
 foreach ( $foo as $bar ) { ...
 ```
 
+When using `case` statements, do it like so:
+
+```php
+
+switch ( $i ) {
+    case 1:
+        echo 'i equals 1';
```
Comment (GaryJones) [inline php-coding-standards.md]: All these `'i equals...'`  should probably be `'$i equals...'`.

File: php-coding-standards.md [hunk 11]
```diff
@@ -147,6 +147,29 @@ Put spaces on both sides of the opening and closing parenthesis of `if`, `elseif
 foreach ( $foo as $bar ) { ...
 ```
 
+When using `case` statements, do it like so:
+
+```php
+
```
Comment (GaryJones) [inline php-coding-standards.md]: Redundant empty line inside the code block. One at the end too.

Comment (ntwb) [inline php-coding-standards.md]: Done

File: php-coding-standards.md (hunk 10 above)
Comment (ntwb) [inline php-coding-standards.md]: Done

Comment (ntwb) [inline php-coding-standards.md]: Also pushed an update to the php.net examples where I snagged this from

https://secure.php.net/manual/en/control-structures.switch.php 😄

**Pull request merged**